### PR TITLE
fix: push files one by one instead of parralel

### PR DESCRIPTION
**Pull request by umutuzgur**

### CODE REVIEW

File: lib/ios-deploy.js [hunk 1]
```diff
@@ -7,7 +7,7 @@ import log from './logger';
 
 const APPLICATION_INSTALLED_NOTIFICATION = 'com.apple.mobile.application_installed';
 const INSTALLATION_STAGING_DIR = 'PublicStaging';
-const APPLICATION_PUSH_TIMEOUT = 60 * 1000;
+const APPLICATION_PUSH_TIMEOUT = 30 * 1000;
```
Comment (mykola-mokhnach) [inline lib/ios-deploy.js]: shouldn't sequential push be slower than the parallel one?

File: lib/ios-deploy.js [hunk 2]
```diff
@@ -64,23 +64,25 @@ class IOSDeploy {
...
           readStream.pipe(writeStream);
+          try {
+            await filePush;
+          } catch (e) {
+            throw new Error(`Couldn't push the file '${itemPath}' within the timeout ${APPLICATION_PUSH_TIMEOUT}ms. Error: ${e.message}`);
+          }
         }
       });
-      try {
-        await B.all(promises).timeout(APPLICATION_PUSH_TIMEOUT);
```
Comment (mykola-mokhnach) [inline lib/ios-deploy.js]: the problem might be, that we were trying to push all files at once. What if we push them chunk by chunk though?

File: lib/ios-deploy.js (hunk 1 above)
Comment (umutuzgur) [reply lib/ios-deploy.js]: This is per file now

File: lib/ios-deploy.js [hunk 3]
```diff
@@ -64,23 +64,25 @@ class IOSDeploy {
...
         const pathOnPhone = path.join(bundlePathOnPhone, path.relative(app, itemPath));
         if (isDir) {
           await afcService.createDirectory(pathOnPhone);
         } else {
-          const readStream = fs.createReadStream(itemPath, {autoClose: true});
-          const writeStream = await afcService.createWriteStream(pathOnPhone, {autoDestroy: true });
-          promises.push(new B((resolve) => writeStream.on('close', resolve)));
+          const readStream = fs.createReadStream(itemPath, { autoClose: true });
+          const writeStream = await afcService.createWriteStream(pathOnPhone, { autoDestroy: true });
+          const filePush = new B((resolve, error) => {
```
Comment (imurchie) [inline lib/ios-deploy.js]: I'd rather keep the same naming as everywhere else (i.e., `resolve` and `reject`, not `error`).

**Pull request merged**